### DISCUSSION TITLE
Error with video_state in OpenAPI definition

### DISCUSSION THREAD

**Discussion by maxalmonte14** [etsy/open-api#661] [Q&A] [answered]
Hi everybody, I've just generated a client from the Etsy Open API specification v3 and I'm running into some problems due to what I consider a non-properly described value. After uploading a video via `/v3/application/shops/{shop_id}/listings/{listing_id}/videos` I should receive a response like the following one, where video_state is a integer going from 0 to 3, as documented

Sample response featured on Etsy's site

```{
"video_id": 1,
"height": 0,
"width": 0,
"thumbnail_url": "string",
"video_url": "string",
"video_state": 0
}
```

OpenAPI specification on Etsy's site

```
"ListingVideo": {
  ...
  "video_state": {
      "type": "string",
      "description": "The current state of a given video. Value is one of `active`, `inactive`, `deleted` or `flagged`.",
      "enum": ["0", "1", "2", "3"],
      "default": 0
  }
}
```

Instead I'm receiving a response like the following, where `video_state` is a string

```
{
   "video_id":1,
   "video_url":"https://sample.mp4",
   "thumbnail_url":"https://sample.jpg",
   "height":270,
   "width":480,
   "video_state":"active"
}
```

I want to believe the Open API definition was supposed to be an enum of strings and not numbers, like this other one from `getListingsByShop`

```
{
  "name": "state",
  ...
  "schema": {
      "type": "string",
      "description": "When _updating_ a listing, this value can be either `active` or `inactive`. Note: Setting a `draft` listing to `active` will also publish the listing on etsy.com. Setting a `sold_out` listing to active will update the quantity to 1 and renew the listing on etsy.com.",
      "enum": [
          "active",
          "inactive",
          "sold_out",
          "draft",
          "expired"
      ],
      "default": "active"
  }
}
```

So could someone please 1) verify that the Open API specification is up to date, and 2) point out to me the string equivalent of 0, 1, 2, and 3 for video state? It seems the states should be `active`, `inactive`, `deleted` and `flagged`, but I don't want to make any assumptions since I don't know the internals of the API.

**Comment by etsyachristensen** [comment]
@maxalmonte14 You are correct. This is an error in our documentation. The numbers you are seeing are the indexes of those values in that same order: 0 = active, 1 = inactive, 2 = deleted, 3 = flagged.

We will get this corrected.

**Comment by etsyachristensen** [comment, marked as answer]
The documentation has been corrected.

**Reply by maxalmonte14** [reply]
> The documentation has been corrected.

Thank you very much good sir!
